mtp_api/apps/transaction/tests/utils.py

```diff
@@ -0,0 +1,79 @@
+import datetime
+import random
+import names
+import string
+
+from django.utils import timezone
+from django.utils.crypto import get_random_string
+
+from prison.models import Prison
+from ..models import Transaction
+
+
+def random_dob():
+    return datetime.date(
+        day=random.randint(1, 29),
+        month=random.randint(1, 12),
+        year=random.randint(1930, 1990)
+    )
+
+
+def random_prison_number():
+    # format: [A-Z]\d{4}[A-Z]{2}
+    return '%s%s%s' % (
+        get_random_string(allowed_chars=string.ascii_uppercase, length=1),
+        get_random_string(allowed_chars=string.digits, length=4),
+        get_random_string(allowed_chars=string.ascii_uppercase, length=2)
+    )
+
+
+def random_reference(prisoner_number=None, prisoner_dob=None):
+    if not prisoner_number or not prisoner_dob:
+        return get_random_string(length=15)
+    return '%s %s' % (
+        prisoner_number.lower(),
+        prisoner_dob.strftime('%d%b%Y').lower()
+    )
+
+
+def generate_transactions(uploads=2, transaction_batch=30):
+    transactions = []
+
+    for upload_counter in range(1, uploads+1):
+        for transaction_counter in range(1, transaction_batch+1):
+            # Records might not have prisoner data and/or might not
+            # have sender data.
+            # Atm, we set the probability of it having either of them
+            # to 80% which is an arbitrary high value as we expect
+            # records to have correct data most of the time.
+            include_prisoner_info = random.randint(0, 100) < 80
+            include_sender_info = random.randint(0, 100) < 80
```

Comment:
This is great :+1:

Much, much clearer about the intent of the variables and how we're using them.